ddl/serial_test.go

```diff
@@ -1158,6 +1159,60 @@ func (s *testSerialSuite) TestAutoRandom(c *C) {
 	})
 }
 
+func (s *testSerialSuite) TestAutoRandomChangeFromAutoInc(c *C) {
```

Comment:
Please add a test case for modifying `INT`/`TINYINT`... to auto_random.

BTW, the current error message is confusing, I think it's better to refine it in this PR:

```
tidb> alter table t modify a int key auto_random;
ERROR 8200 (HY000): can't change column constraint - 1
```